### PR TITLE
API-11324 VFQ: improve appointment status mapping for SCHEDULED/KEPT

### PR DESCRIPTION
**Pull request by taylorrassmannlibertyits**
https://vajira.max.gov/browse/API-11324
Formatting xml docs and added SCHEDULED/KEPT appointment status proce…ssing.

### CODE REVIEW

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4AppointmentTransformer.java [hunk 1]
```diff
@@ -30,21 +31,19 @@
...
-
     switch (appointmentStatus) {
       case "NO-SHOW":
         return AppointmentStatus.noshow;
       case "CANCELLED":
         return AppointmentStatus.cancelled;
-        // Intentionally falling through for SCHEDULED/KEPT and NO ACTION TAKEN
       case "SCHEDULED/KEPT":
+        return bookedOrFulfilledStatus(dateTime);
+        // Intentionally falling through NO ACTION TAKEN
```
Comment (jblefkove-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4AppointmentTransformer.java:46]: this comment would be clearer under the `NO ACTION TAKEN` case

**Review by jhulbert-va — COMMENTED** [review]
Looks good to me. We're holding off on VFQ merges right now though because it's still failing to build.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4AppointmentTransformer.java [hunk 2]
```diff
@@ -30,22 +31,20 @@
...
-  private Appointment.AppointmentStatus appointmentStatus(ValueOnlyXmlAttribute status) {
-
+  private AppointmentStatus appointmentStatus(ValueOnlyXmlAttribute status, String dateTime) {
     if (status == null) {
       return null;
     }
-
     String appointmentStatus = status.value();
-
     switch (appointmentStatus) {
```
Comment (jblefkove-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4AppointmentTransformer.java:39]: Potential NPE here if status is non-null, but `status.value()` is null?

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**